src/pydap/handlers/dap.py

```diff
@@ -228,6 +244,7 @@ def add_dap2_proxies(self):
             var.data = SequenceProxy(
                 self.base_url,
                 template,
+                selection=self.selection,
```

Comment:
This was completely missing even for DAP2. Now it is being passed and constraints expression for sequences work as expected (DAP2).